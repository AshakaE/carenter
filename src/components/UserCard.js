import React, { useState } from 'react';
import { useHistory, Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import UserForm from './forms/UserForm';
import { userDelete } from '../assets/getAuth';
import styles from '../assets/css/user.module.css';
import form from '../assets/css/login.module.css';

const UserCard = (props) => {
  const [edit, SetEdit] = useState(false);
  const history = useHistory();
  const {
    user,
  } = props;

  const handleEdit = () => {
    SetEdit(!edit);
  };

  const handleDelete = async () => {
    await userDelete();
    history.push('/');
  };

  return (
    <>
      <div className={styles.container}>
        <div className={styles.items}>
          <div className={form.header}>{user}</div>
          <UserForm edit={edit} />
          <button type="button" onClick={handleEdit} className={styles.btn}>
            Update login information
          </button>
          <button
            type="button"
            onClick={handleDelete}
            className={styles.btnDel}
          >
            Delete account
          </button>
          <div className={styles.links}>
            <div className={styles.link}>
              <Link to="/cars">Cars gallery</Link>
            </div>
            <div className={styles.link}>
              <Link to="/bookings">My Bookings</Link>
            </div>
          </div>
        </div>
      </div>
    </>
  );
};

UserCard.propTypes = {
  user: PropTypes.string,
};

UserCard.defaultProps = {
  user: [],
};

export default UserCard;
