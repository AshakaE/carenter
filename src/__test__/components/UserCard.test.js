import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import UserCard from '../../components/UserCard';

describe('UserCard', () => {
  it('renders correctly', () => {
    const test = 'user';
    const tree = renderer
      .create(
        <BrowserRouter>
          <UserCard
            user={test}
          />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
