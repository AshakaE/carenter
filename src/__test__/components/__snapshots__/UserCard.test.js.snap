// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="items"
  >
    <div
      className="header"
    >
      user
    </div>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      Update login information
    </button>
    <button
      className="btnDel"
      onClick={[Function]}
      type="button"
    >
      Delete account
    </button>
    <div
      className="links"
    >
      <div
        className="link"
      >
        <a
          href="/cars"
          onClick={[Function]}
        >
          Cars gallery
        </a>
      </div>
      <div
        className="link"
      >
        <a
          href="/bookings"
          onClick={[Function]}
        >
          My Bookings
        </a>
      </div>
    </div>
  </div>
</div>
`;
